File: benchmark/2026-09-17-7677581(latest)/semantic-compression/witnesses/swift/chunk_B/f10p2_V6_user_rhs.swift
struct Money { var cents: Swift.Int }
func + (l: Swift.Double, r: Money) -> Money { print("user +"); return Money(cents: r.cents + Swift.Int(l)) }

let i: Int32 = 3
let d = Money(cents: 40)
// BEGIN PROBE F10.P2
let sum = Double(i) + d
// END PROBE F10.P2
print(sum.cents)

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/witnesses/rust/chunk_B/f10p2_V6_userrhs.rs
use std::ops::Add;

struct D(f64);
impl Add<D> for f64 { type Output = f64; fn add(self, o: D) -> f64 { println!("user add"); self * o.0 } }

fn main() {
    let i: i32 = 3;
    let d: D = D(0.5);
// BEGIN PROBE F10.P2
    let sum = i as f64 + d;
// END PROBE F10.P2
    println!("{:?}", sum);
}
